controllers/api_controller.js

```diff
@@ -1,18 +1,39 @@
-const {fetchTopics} = require('../models/api_model')
-const fs = require('fs/promises')
+const { fetchTopics, fetchArticleById } = require("../models/api_model");
+const fs = require("fs/promises");
+const endPoints = require("../endpoints.json");
 
 exports.getTopics = (req, res, next) => {
-    fetchTopics().then((topics) => {
-        res.status(200).send({topics})
+  fetchTopics()
+    .then((topics) => {
+      res.status(200).send({ topics });
     })
     .catch((err) => {
-        next(err)
-    })
-}
+      next(err);
+    });
+};
 
 exports.getApi = (req, res, next) => {
-   return fs.readFile('./endpoints.json', 'utf-8').then((data) => {
-        const endpoint = JSON.parse(data)
-        res.status(200).send(endpoint)
-    }) 
-}
\ No newline at end of file
+  res
+    .status(200)
+    .send(endPoints)
+
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.getArticleById = (req, res, next) => {
+  const { article_id } = req.params;
+  fetchArticleById(article_id).then((article) => {
+    if (!article) {
+      return res.status(404).send({ msg: "Article not found" });
```

Comment:
we actually want to return a Promise rejection!  All of this code actually sending an error needs to be in some error handling middleware.